tools/parse-release-notes

```diff
@@ -0,0 +1,126 @@
+const DEFAULT_NOTE_DE = 'Wir haben hinter den Kulissen hart gearbeitet, um sicherzustellen, dass alles so funktioniert, wie es soll. Wenn Sie bemerken, dass etwas nicht funktioniert, lassen Sie es uns wissen!\n'
+
+const PLATFORM_ANDROID = 'android'
+const PLATFORM_IOS = 'ios'
+const PLATFORM_WEB = 'web'
+
+program
+  .option('-d, --debug', 'enable extreme logging')
+  .option('--ios', 'include release notes for ios')
+  .option('--android', 'include release notes for android')
+  .option('--web', 'include release notes for web. may not be used with --android or --ios flags')
+  .option('--production', 'whether to hide extra information, e.g. issue keys, hidden notes and platforms and prepare the notes for a store. may not be used with multiple platforms.')
+  .option('--destination <destination>', 'if specified the parsed notes are saved to the directory')
+  .requiredOption('--source <source>', 'the directory of the release notes to parse', `${RELEASE_NOTES_DIR}/${UNRELEASED_DIR}`)
+  .requiredOption('--language <language>', 'the language of the release notes to parse', DEFAULT_LANGUAGE)
+
+const formatNotes = ({ notes, language, production, platformName }) => {
+  if (notes.length === 0) {
+    return ''
+  }
+
+  const formattedNotes = notes
+    .map(note => {
+      const localizedNote = note[language] || note[DEFAULT_LANGUAGE]
+      return production ? `* ${localizedNote}` : `* ${note.issue_key}: ${localizedNote}`
+    })
+    .join('\n')
+
+  return platformName ? `\n${platformName}:\n${formattedNotes}` : `${formattedNotes}`
+}
+
+const isNoteRelevant = ({ note, platforms }) => platforms.some(platform => note.platforms.includes(platform))
+const isNoteCommon = ({ note, platforms }) => platforms.every(platform => note.platforms.includes(platform))
+
+const parseReleaseNotes = ({ language, source, android, ios, web, production }) => {
+  const platforms = []
+  if (android) {
+    platforms.push(PLATFORM_ANDROID)
+  }
+  if (ios) {
+    platforms.push(PLATFORM_IOS)
+  }
+  if (web) {
+    platforms.push(PLATFORM_WEB)
+  }
+
+  if (platforms.length === 0) {
+    throw new Error('No platforms selected! Use --ios, --android and --web flags.')
+  } else if (web && (ios || android)) {
+    throw new Error('Usage of --ios or --android together with --web flag is not supported.')
+  } else if (platforms.length > 1 && production) {
+    throw new Error('Usage of multiple platforms in production mode is not supported.')
+  }
+
+  const fileNames = fs.readdirSync(source)
+
+  // Load all notes not belonging to a release
+  const relevantNotes = fileNames
+    .filter(fileName => fileName !== GITKEEP_FILE)
+    .map(fileName => yaml.safeLoad(fs.readFileSync(`${source}/${fileName}`)))
+    .filter(note => isNoteRelevant({ note, platforms }))
+
+  // If the production flag is set, hide information that is irrelevant for users
+  if (production) {
+    const productionNotes = relevantNotes.filter(note => note.show_in_stores)
```

Comment:
What do you think about this:

IF there are only non-production notes THEN show a random non-production note instead of the default note.

That way we avoid generating the same release notes.

Reply:
I don't really like it to be honest.
Firstly, there may be no non production notes as well and this just adds more complexcity to the script.
Secondly, Non production notes are probably quite technically and not user related at all. I think users could be really confused if there is suddenly a release note saying e.g. Added design for Malte or Fixed some stuff in the CI.

Maybe we could rather add a production and a non production release notes for technical tasks? E.g. if working on the CI:
We made sure that you are always getting your app updates in time vs Fixed failing tests in the CI.

Also, we could add more default texts and pick one randomly. I would do this in a separate task though. If you like the idea could you create an issue?